invoice/views.py

```diff
@@ -71,17 +51,49 @@ class BankAccountListView(ListView):
 
 class CustomerCreateView(CreateView):
     """Create a new customer."""
+    template_name = 'invoice/customer_form.html'
     model = Customer
-    fields = '__all__'
+    fields = ['first_name', 'last_name', 'email']
     success_url = reverse_lazy('customer-list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['address_form'] = AddressForm(self.request.POST)
+        else:
+            context['address_form'] = AddressForm()
+        return context
+
+    def form_valid(self, form):
+        """Create a new customer and a new address."""
+        address_form = AddressForm(self.request.POST)
+        address = address_form.save()
+        customer = form.save(commit=False)
+        customer.address = address
+        return super().form_valid(form)
+
 
 class CustomerUpdateView(UpdateView):
     """Update an existing customer."""
-    model = Customer
-    fields = '__all__'
+    template_name = 'invoice/customer_form.html'
+    form_class = CustomerForm
     success_url = reverse_lazy('customer-list')
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.request.POST:
+            context['address_form'] = AddressForm(self.request.POST)
+        else:
+            context['address_form'] = AddressForm(instance=self.object.address)
+        return context
+
+    def form_valid(self, form):
+        """Updates an existing vendor including the address and the bank account."""
+        address_form = AddressForm(instance=self.object.address, data=self.request.POST)
+        if address_form.is_valid():
+            address_form.save()
```

Comment:
what about error handling?
what happens when the address_form is invalid?

Reply:
We should handle it via messages, which are to be implemented.

Reply:
Correction, I use the build in invalid method